sources/api/migration/migrations/v1.21.0/k8s-reserved-cpus-v0-1-0/src/main.rs

```diff
@@ -0,0 +1,18 @@
+use migration_helpers::common_migrations::AddSettingsMigration;
+use migration_helpers::{migrate, Result};
+use std::process;
+
+/// Add the option to set Kubernetes reserved-cpus
+fn run() -> Result<()> {
+    migrate(AddSettingsMigration(&["settings.kubernetes.reserved-cpus"]))
```

Comment:
Don't we need to limit this for k8s only? And have a conditional here to perform a `Noop` migration?

Reply:
I couldn't find any example of that for an "Add a new setting" migration.

I found

https://github.com/bottlerocket-os/bottlerocket/blob/1.20.x/sources/api/migration/migrations/v1.20.0/add-ntp-default-options-v0-1-0/src/main.rs

and

https://github.com/bottlerocket-os/bottlerocket/blob/1.19.x/sources/api/migration/migrations/v1.19.2/add-ecs-enable-container-metadata/src/main.rs

and

https://github.com/bottlerocket-os/bottlerocket/blob/1.19.x/sources/api/migration/migrations/v1.19.0/add-additional-ecs-settings/src/main.rs

all of which do the same as mine...

Reply:
https://github.com/bottlerocket-os/bottlerocket/blob/a1bcca5556c9f0053df16aabf36920adae4e6944/sources/api/migration/migrations/archived/v1.15.0/oci-defaults-docker-setting/src/main.rs#L2 is the inverse of what you need here.

```suggestion
     if cfg!(variant_runtime = "k8s") {
         migrate(AddSettingsMigration(&["settings.kubernetes.reserved-cpus"]))
    } else {
        migrate(NoOpMigration);
    }
```

The example I linked to used `?` but I don't think it is required.

Reply:
I tried a few conbinations of this and the other change you linked to.

I'm stuck on...

```
  #19 0.466 7  | /     if cfg!(variant_runtime = "k8s") {                                                                                                             
  #19 0.466 8  | |         migrate(AddSettingsMigration(&["settings.kubernetes.reserved-cpus"]))
  #19 0.466    | |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `()`, found `Result<(), Error>`                             
  #19 0.466 9  | |     } else {                                                                                                                                       
  #19 0.466 10 | |        migrate(NoOpMigration);                                                                                                                     
  #19 0.466 11 | |     }                                                                                                                                              
  #19 0.466    | |_____- expected this to be `()`                                                                                                                     
  #19 0.466    |
  #19 0.466    = note: expected unit type `()`
  #19 0.466                    found enum `Result<(), migration_helpers::error::Error>`
  #19 0.466 help: consider using a semicolon here
  #19 0.466    |
  #19 0.466 8  |         migrate(AddSettingsMigration(&["settings.kubernetes.reserved-cpus"]));
  #19 0.466    |                                                                              +
  #19 0.466 help: consider using a semicolon here
  #19 0.466    |
```

... any hints?

Reply:
Oh, sorry, there was a typo in my suggestion, there should not be a ; at the end of the first migration call to end the statement.
```suggestion
 if cfg!(variant_runtime = "k8s") {
         migrate(AddSettingsMigration(&["settings.kubernetes.reserved-cpus"]))
    } else {
        migrate(NoOpMigration)
    }
```
The error is complaining about mixing and matching statements vs an expression. In this case, if you leave off the `;` it will return the results of the function call, in this case `migration()` so want both statements to not have `;` but then you also don't want `Ok(())` at the end of your `run()` function. So the code I used to get this to compile looks like this for the entire file:
```
use migration_helpers::common_migrations::{AddSettingsMigration, NoOpMigration};
use migration_helpers::{migrate, Result};
use std::process;

/// Add the option to set Kubernetes reserved-cpus
fn run() -> Result<()> {
     if cfg!(variant_runtime = "k8s") {
         migrate(AddSettingsMigration(&["settings.kubernetes.reserved-cpus"]))
    } else {
        migrate(NoOpMigration)
    }
}

// Returning a Result from main makes it print a Debug representation of the error, but with Snafu
// we have nice Display representations of the error, so we wrap "main" (run) and print any error.
// https://github.com/shepmaster/snafu/issues/110
fn main() {
    if let Err(e) = run() {
        eprintln!("{}", e);
        process::exit(1);
    }
}
```

Reply:
Either this suggestion, or adding `?;` and `Ok(())` at the end should be enough.